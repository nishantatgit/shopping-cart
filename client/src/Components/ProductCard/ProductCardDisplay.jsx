import React from 'react';
import {Link} from 'react-router-dom';

export const ProductCardDisplay = ({product, addToCartHandler}) => {
    return(
        <section key={product.pid} className="col-6 col-md-4 p-2">
            <div className="card h-100">
                <img className="card-img-top" src={`${window.location.origin}/imgs/${product.pid}.jpg`} alt={`image of ${product.pname}`} title={product.pname}/>
                <div className="card-body d-flex flex-column">
                    <div className="card-title"><span>{product.pname}</span><span className="font-weight-bold float-right mr-1">${product.inventories.length && product.inventories[0].price}</span></div>
                    <p className="text-secondary pb-2">{product.pdesc}</p>
                    <div className="d-flex justify-content-center mt-auto">
                        <Link to={`/product-detail/${product.pid}`} className="btn mr-3 bg-info text-white">More Info</Link>
                        <button className="btn  float-right bg-info text-white" onClick={addToCartHandler}>Add To Cart</button>
                    </div>
                </div>
            </div>
        </section>
    );
}
