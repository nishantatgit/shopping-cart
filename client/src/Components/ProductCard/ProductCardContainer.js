import React from 'react';
import {connect} from 'react-redux';
import {ProductCardDisplay} from './ProductCardDisplay.jsx';

class ProductCardContainer extends React.Component{
    render(){
        let {product,addToCartHandler} = this.props;
        return <ProductCardDisplay product={product} addToCartHandler={addToCartHandler.bind(undefined,product)}/>;
    }
}
export  default connect()(ProductCardContainer);