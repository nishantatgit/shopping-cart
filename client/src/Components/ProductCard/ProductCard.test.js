import React from 'react';
import {shallow} from 'enzyme';
import {ProductCardDisplay} from './ProductCardDisplay.jsx';
import ProductCardContainer from './ProductCardContainer';

describe('ProductCardDisplay', () => {
    it('should render correctly', () => {
        const product = {
            pid : 'p-1',
            pname: 'test-1',
            inventories : { price : 1 }
        }
        const component = shallow(<ProductCardDisplay product={product} />);
        expect(component).toMatchSnapshot();
    });
});

