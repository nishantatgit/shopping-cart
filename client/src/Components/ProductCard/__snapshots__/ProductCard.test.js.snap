// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardDisplay should render correctly 1`] = `
<section
  className="col-6 col-md-4 p-2"
  key="p-1"
>
  <div
    className="card h-100"
  >
    <img
      alt="image of test-1"
      className="card-img-top"
      src="http://localhost/imgs/p-1.jpg"
      title="test-1"
    />
    <div
      className="card-body d-flex flex-column"
    >
      <div
        className="card-title"
      >
        <span>
          test-1
        </span>
        <span
          className="font-weight-bold float-right mr-1"
        >
          $
        </span>
      </div>
      <p
        className="text-secondary pb-2"
      />
      <div
        className="d-flex justify-content-center mt-auto"
      >
        <Link
          className="btn mr-3 bg-info text-white"
          replace={false}
          to="/product-detail/p-1"
        >
          More Info
        </Link>
        <button
          className="btn  float-right bg-info text-white"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </div>
</section>
`;
